tensorly/datasets/nn_parafac_IL2_example.py

```diff
@@ -0,0 +1,89 @@
+"""
+Non-negative PARAFAC Decomposition of IL-2 Response Data
+==============================
+
+Here we will provide an example of how to use non-negative PARAFAC tensor 
+decomposition (:func:`tensorly.decomposition.parafac`) to both reduce the dimensionality 
+of a tensor of experimental data, as well as make insights regarding the underlying structure 
+of that data.
+
+To do this, we will work with a tensor of experimentally measured cell signaling data.
+"""
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+from tensorly.decomposition import non_negative_parafac
+from imports import IL2data
+
+#%%
+# Here we will load a tensor of experimentally measured cellular responses to 
+# IL-2 stimulation. IL-2 is a naturally occurring immune signaling molecule 
+# which has been engineered and mutated in attempts to alter its signaling activity, 
+# and increase or decrease its interactions with particular cell types. 
+# Here, our data contains the responses of 8 different cell types to 13 different 
+# IL-2 mutants, at 4 different timepoints, at 12 standardized IL-2 concentrations.
+# 
+# Therefore, our tensor will have shape (13 x 4 x 12 x 8), with dimensions
+# representing IL-2 mutant, stimulation time, dose, and cell type respectively.
+
+
+responseData = IL2data()
+IL2mutants, cells = responseData.ligands, responseData.cells
+print(responseData.tensor.shape, responseData.dims)
+
+#%%
+# IL-2 mutants, at 4 different timepoints, at 12 standardized IL-2 concentrations.
+# Now we will run non-negative PARAFAC tensor decomposition to reduce the dimensionality 
+# of our tensor. We will use 3 components, and normalize our resulting tensor to aid in 
+# future comparisons of correlations across components.
+
+
+tensorFact = non_negative_parafac(np.nan_to_num(responseData.tensor), init='random', rank=3, mask=np.isfinite(responseData.tensor), n_iter_max=5000, tol=1e-9, random_state=1)
+tensorFact.normalize()
```

Comment:
Perhaps we should call explicitly the normalize function from the module on the factorization rather than use its method?